tests/experiments/experiment_unittests.py

```diff
@@ -245,43 +268,100 @@ def experiment_test_generator(module_name):
 
+    # Read in config.ini file for current site to make necessary directories
+    path = f'{os.environ["BOREALISPATH"]}/config/' \
+           f'{os.environ["RADAR_ID"]}/' \
+           f'{os.environ["RADAR_ID"]}_config.ini'
+    try:
+        with open(path, 'r') as data:
+            raw_config = json.load(data)
+    except OSError:
+        errmsg = f'Cannot open config file at {path}'
+        raise ValueError(errmsg)
+
+    # These directories are required for ExperimentHandler to run
+    data_directory = raw_config["data_directory"]
+    log_directory = raw_config["log_directory"]
+    hdw_path = raw_config['hdw_path']
+    hdw_dat_file = f'{hdw_path}/hdw.dat.{os.environ["RADAR_ID"]}'
+    if not os.path.exists(data_directory):
+        os.makedirs(data_directory)
+    if not os.path.exists(log_directory):
+        os.makedirs(log_directory)
+    if not os.path.exists(hdw_path):
+        os.makedirs(hdw_path)
+    if not os.path.exists(hdw_dat_file):
+        open(hdw_dat_file, 'w')
+
+    experiments = args.experiments
+    print(f"Running tests on experiments {experiments}")
     if experiments is None:  # Run all unit tests and experiment tests
         build_unit_tests()
         build_experiment_tests()
-        unittest.main(argv=sys.argv[:1])
+        argv = None
     else:  # Only test specified experiments
-        build_experiment_tests()
+        build_experiment_tests(experiments, args.kwargs)
         exp_tests = []
         for exp in experiments:
-            # Check experiment exists 
+            # Check experiment exists
             if hasattr(TestActiveExperiments, f"test_{exp}"):
                 # Create correct string to test the experiment with unittest
                 exp_tests.append(f"TestActiveExperiments.test_{exp}")
             else:
                 print(f"Could not find experiment {exp}. Exiting...")
                 exit(1)
-        unittest.main(argv=sys.argv[:1] + exp_tests)
+        argv = [parser.prog] + exp_tests
+    if print_results:
+        result = unittest.main(module=args.module, argv=argv, exit=False, buffer=buffer)
+    else:
+        result = redirect_to_devnull(unittest.main, module=args.module, argv=argv, exit=False, buffer=buffer)
+
+    # Clean up the directories/files we created
+    try:
+        os.removedirs(data_directory)
+    except OSError:     # If directories not empty, this will fail. That is fine.
+        pass
+    try:
+        os.removedirs(log_directory)
+    except OSError:     # If directories not empty, this will fail. That is fine.
+        pass
+    if os.path.getsize(hdw_dat_file) == 0:
+        try:
+            os.remove(hdw_dat_file)
+        except OSError:     # Path is a directory
+            os.removedirs(hdw_dat_file)
+        else:   # File removed, now clean up directories
+            try:
+                os.removedirs(hdw_path)
+            except OSError:     # If directories not empty, this will fail. That is fine.
+                pass
```

Comment:
Can this all be combined into a single try: except block?

Reply:
I wanted to keep these as separate blocks so that if the first thing fails, it still tries to remove subsequent things. If it's all in one big block it will skip everything as soon as an exception is thrown